src/taskActions.js

```diff
@@ -0,0 +1,63 @@
+import storage from './storage.js';
+
+const taskActions = (() => {
+  const addTask = (description) => {
+    const tasks = storage.get('tasks');
+    const task = {
+      index: tasks.length + 1,
+      completed: false,
+      description,
+    };
+    tasks.push(task);
+    storage.set('tasks', tasks);
+    return task;
+  };
+
+  const updateTask = (index, description) => {
+    const tasks = storage.get('tasks');
+    const task = tasks[index - 1];
+    task.description = description;
+    storage.set('tasks', tasks);
+    return { task, tasks };
+  };
+
+  const completeTask = (index, checked) => {
+    const tasks = storage.get('tasks');
+    const task = tasks[index - 1];
+    task.completed = checked;
+    storage.set('tasks', tasks);
+    return { task, tasks };
+  };
+
+  const deleteTask = (index) => {
+    let tasks = storage.get('tasks');
+    tasks.splice(index - 1, 1);
+    tasks = tasks.map((task, i) => {
+      task.index = i + 1;
+      return task;
+    });
+    storage.set('tasks', tasks);
+    return tasks;
+  };
```

Comment:
- [ ] [OPTIONAL] Good job adding different functions for the CRUD operation, but it seems that there is an issue with your delete function, sometimes I have to click twice on the delete button to reflex the changes, it could be a good idea to fix this issue 😉